scipy/optimize/optimize.py

```diff
@@ -2102,7 +2105,10 @@ def _minimize_scalar_golden(func, brack=None, args=(),
     f1 = func(*((x1,) + args))
     f2 = func(*((x2,) + args))
     funcalls += 2
-    while (numpy.abs(x3 - x0) > tol * (numpy.abs(x1) + numpy.abs(x2))):
+    nit = 0
+    for i in xrange(maxiter):
+        if (numpy.abs(x3 - x0) <= tol * (numpy.abs(x1) + numpy.abs(x2))):
```

Comment:
Indeed, you reach a point where _gr*x2+_gC*x3 equals x2 or x3 because there are no floating-point numbers left between x2 and x3 (or the analogous situation in the other branch). This occurs in the original situation that triggered the bug, and it's worth catching because no further progress will ever be made.